### PR TITLE
Implement axis locking options for rigid bodies

### PR DESCRIPTION
**Pull request by ibtehajn**
Linear and angular motion of rigid bodies can now be restricted along
specific world-space axes.

Signed-off-by: Ibtehaj Nadeem <81370835+ibtehajn@users.noreply.github.com>

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/Physics/Configuration/RigidBodyConfiguration.h [hunk 1]
```diff
@@ -62,6 +62,16 @@ namespace AzPhysics
         bool m_computeInertiaTensor = true;
         bool m_computeMass = true;
 
+        //! Flags to restrict motion along specific world-space axes.
```
Comment (hultonha) [inline Code/Framework/AzFramework/AzFramework/Physics/Configuration/RigidBodyConfiguration.h:65]: nit: This likely won't come out correctly in doxygen (the comment will just be applied to the first member) you might want to check the @group annotation (or just make this a regular `//` comment) - same below

**Review by hultonha — APPROVED** [review]
LGTM! Thanks for adding the test too 🙂

File: Code/Framework/AzFramework/AzFramework/Physics/Configuration/RigidBodyConfiguration.h (hunk 1 above)
Comment (ibtehajn) [reply Code/Framework/AzFramework/AzFramework/Physics/Configuration/RigidBodyConfiguration.h:65]: Good point, will fix.

**Comment by ibtehajn** [comment]
> 
> 
> LGTM! Thanks for adding the test too 🙂

The test was mostly just taken from @SergeyAMZN PR  and adjusted a bit - credit goes to him!

**Review by SergeyAMZN — APPROVED** [review]
looks good! 
wondering why is it named "linear momentum" when physx doc just says "motion along" & "motion around"?

**Comment by ibtehajn** [comment]
> 
> 
> looks good!
> wondering why is it named "linear momentum" when physx doc just says "motion along" & "motion around"?

Good point - I think "motion along" and "motion around" are clearer (and more correct). Will update.

**Review by moraaar — APPROVED** [review]

File: Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp [hunk 2]
```diff
@@ -156,6 +156,33 @@ namespace PhysX
                     ->DataElement(AZ::Edit::UIHandlers::Default, &AzPhysics::RigidBodyConfiguration::m_kinematic,
                         "Kinematic", "Rigid body is kinematic")
                         ->Attribute(AZ::Edit::Attributes::Visibility, &AzPhysics::RigidBodyConfiguration::GetKinematicVisibility)
+
+                    // Linear axis locking properties
+                    ->ClassElement(AZ::Edit::ClassElements::Group, "Linear Axis Locking")
+                        ->Attribute(AZ::Edit::Attributes::AutoExpand, false)
```
Comment (AMZN-AlexOteiza) [inline Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp:162]: nit: I don't think auto-expanding is really nice in terms of UX, out component card is already long..

**Review by AMZN-AlexOteiza — APPROVED** [review]

File: Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp (hunk 2 above)
Comment (ibtehajn) [reply Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp:162]: Auto expand is set to `false`.

**Pull request merged**